### PR TITLE
Document supported join types

### LINKED ISSUE #717: Change required: add documents for all join types.
### Related code PR

We currently support NATURAL JOIN in both streaming and batch mode. But for CROSS JOIN, we only support it in the batch mode. It's disallowed in the streaming mode because of the expensive cost of nested-loop join.

### Which part(s) of the docs might be affected or should be updated? And how?

Typically, databases have a seperated doc page named 'Joins' that describes the supported join types. 
Examples:
- https://materialize.com/docs/sql/join/
- https://www.cockroachlabs.com/docs/stable/joins.html
- https://crate.io/docs/crate/reference/en/5.2/concepts/joins.html
- https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/joins/

The content mentioned above should be included in this page.

### Reference

_No response_

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

  [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #717

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/query-syntax/query-syntax-join-clause.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+```sql
+<table_expression> FULL [ OUTER ] JOIN <table_expression> ON <join_conditions>;
+<table_expression> FULL [ OUTER ] JOIN <table_expression> USING (<col_name>, <col_name>, ...);
+<table_expression> NATURAL FULL [ OUTER ] JOIN <table_expression>;
+```
+
+## Windows joins
+
+You can join two sources that have the same watermarks. This is called a window join. In a regular join (that is, a join without time attributes), the join state may grow without restriction. With window joins, the join state size can be kept at a resonable size.
```
Comment (lmatz) [inline docs/sql/query-syntax/query-syntax-join-clause.md]: By `same watermarks`, does it mean that ` - INTERVAL '1' SECOND` part must be the same?

**Review by lmatz — APPROVED** [review]
rest LGTM

File: docs/sql/query-syntax/query-syntax-join-clause.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+```sql
+CREATE SOURCE t1 (ts1 timestamp with time zone, a1 int, b1 int, WATERMARK for ts1 AS ts1 - INTERVAL '1' SECOND) WITH (
+    connector = 'datagen',
+    datagen.rows.per.second = '10',
+);
+CREATE SOURCE t2 (ts2 timestamp with time zone, a2 int, b2 int, WATERMARK for ts2 as ts2 - INTERVAL '1' SECOND) WITH (
+    connector = 'datagen',
+    datagen.rows.per.second = '10',
+);
```
Comment (chenzl25) [inline docs/sql/query-syntax/query-syntax-join-clause.md]:
```suggestion
CREATE SOURCE s1 (
	id int, 
	value int, 
	ts TIMESTAMP, 
	WATERMARK FOR ts AS ts - INTERVAL '20' SECOND
) WITH (connector = 'datagen');


CREATE SOURCE s2 (
	id int, 
	value int, 
	ts TIMESTAMP, 
	WATERMARK FOR ts AS ts - INTERVAL '20' SECOND
) WITH (connector = 'datagen');
```

File: docs/sql/query-syntax/query-syntax-join-clause.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+```
+
+You can join them with the following statement:
+
+```sql
+SELECT * FROM t1
+JOIN t2
+ON ts1 = ts2 
+AND a1 = a2;
```
Comment (chenzl25) [inline docs/sql/query-syntax/query-syntax-join-clause.md]:
```suggestion
CREATE MATERIALIZED VIEW window_join AS
SELECT s1.id AS id1,
       s1.value AS value1,
       s2.id AS id2,
       s2.value AS value2
FROM TUMBLE(s1, ts, interval '1' MINUTE)
JOIN TUMBLE(s2, ts, interval '1' MINUTE)
ON s1.id = s2.id and s1.window_start = s2.window_start;
```

File: docs/sql/query-syntax/query-syntax-join-clause.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+In an interval join, the `interval_condtion` must be a watermark-based range.
+
+For example, for sources `t1` and `t2` used in the above section, you can create an interval join:
+
+```sql
+SELECT * FROM t1
+JOIN t2 
+WHERE a1 = a2 
+AND ts1 BETWEEN ts2 - INTERVAL '10' SECOND AND ts2 + INTERVAL '15' SECOND;
```
Comment (chenzl25) [inline docs/sql/query-syntax/query-syntax-join-clause.md]:
```suggestion
CREATE MATERIALIZED VIEW interval_join AS
SELECT s1.id AS id1,
       s1.value AS value1,
       s2.id AS id2,
       s2.value AS value2
FROM s1 JOIN s2
ON s1.id = s2.id and s1.ts between s2.ts and s2.ts + INTERVAL '1' MINUTE;
```

File: docs/sql/query-syntax/query-syntax-join-clause.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+The syntax of an interval join is:
+
+```sql
+<table_expression> JOIN <table_expression> ON <equality_join_condition> AND <interval_condition> ...;
+```
+
+In an interval join, the `interval_condtion` must be a watermark-based range.
+
+For example, for sources `t1` and `t2` used in the above section, you can create an interval join:
```
Comment (chenzl25) [inline docs/sql/query-syntax/query-syntax-join-clause.md]:
```suggestion
For example, for sources `s1` and `s2` used in the above section, you can create an interval join:
```

**Review by chenzl25 — COMMENTED** [review]
I think we can use a better example to illustrate window join and interval join.

File: docs/sql/query-syntax/query-syntax-join-clause.md (hunk 1 above)
Comment (chenzl25) [reply docs/sql/query-syntax/query-syntax-join-clause.md]: The watermarks of these two sources don't need to be the same, but typically they should use the same windowing table-valued functions (e.g. `TUMBLE`). Watermark columns are used to clean the state after the window is closed.

**Pull request merged**